components/match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua

```diff
@@ -89,55 +127,102 @@ function StarcraftMatchGroupInput._readDate(matchArgs)
 end
 
 ---@param match table
----@return table
-function StarcraftMatchGroupInput._checkFinished(match)
-	if Logic.readBoolOrNil(match.finished) == false then
-		match.finished = false
-		return match
-	elseif Logic.readBool(match.finished) then
-		match.finished = true
-	elseif Logic.isNotEmpty(match.winner) then
-		match.finished = true
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	local maps = {}
+	local subGroup = 0
+	for mapKey, rawMapInput, mapIndex in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local mapInput = Json.parseIfString(rawMapInput)
+		local map
+		map, subGroup = MapFunctions.readMap(mapInput, subGroup, #opponents)
+
+		map.participants = MapFunctions.getParticipants(mapInput, opponents)
+
+		MapFunctions.getModeAndEnrichExtradata(map, mapInput, map.participants, opponents)
+
+		map.vod = Logic.emptyOr(mapInput.vod, match['vodgame' .. mapIndex])
+
+		table.insert(maps, map)
+		match[mapKey] = nil
 	end
 
-	-- Match is automatically marked finished upon page edit after a
-	-- certain amount of time (depending on whether the date is exact)
-	if match.finished ~= true then
-		local threshold = match.dateexact and 30800 or 86400
-		if match.timestamp + threshold < NOW then
-			match.finished = true
-		end
+	return maps
+end
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInput.computeMatchScoreFromMapWinners(maps, opponentIndex)
 	end
+end
 
-	return match
+---@param opponent table
+function MatchFunctions.addOpponentExtradata(opponent)
+	opponent.extradata = Table.merge(opponent.extradata or {}, {
+		advantage = tonumber(opponent.advantage),
+		penalty = tonumber(opponent.penalty),
+		score2 = opponent.score2,
+		isarchon = opponent.isarchon,
+	})
```

Comment:
the table merge is intended for the case that we get the substitutes stuff in TC
because then opp.extradata is not necessarily empty for team opponents

Reply:
table merge is still there 

https://github.com/Liquipedia/Lua-Modules/pull/4550#discussion_r1723156152